### DISCUSSION TITLE
dashboard with dropdown selection of host

### DISCUSSION THREAD

**Discussion by levp7** [jritter/openscap-report-publisher#9] [Ideas]
Hi, thanks for you SCAPinoculars work.  I've got your dashboard working.  I'm quite new to github and Grafana.  I was wondering if you could perhaps consider adding host (Target) drop down selection and filtering ?
Also sorting the rules by severity order - High, Medium, Low, Unknown ? and colour Red, Orange, Yellow, blue respectively ?
Thanks again.

**Comment by jritter** [comment]
Hi, 

Thanks for your feedback, great to hear that you got it to work! Yeah, improving the dashboard experience is definitely high on my todo list, although I have to say that I have limited time to work on this. 

Thanks for your patience

**Comment by jritter** [comment]
Hi @levp7 ,

I made some tweaks in the Grafana Dashboard, which should cover your suggestions above. Could you give it a go, and let me know what you think?

Thanks

**Reply by levp7** [reply]
Hi @jritter, 

Thank you for taking into consideration my suggestions and updating your dashboard.

I have since used it and it seems to be working well.

At first, I had issues and couldn't see any data on the dashboard.  On looking at the settings, I noticed the variables not configured and not pointing to Prometheus.  I thought this to be odd, so I looked at the dashboard json and could see it there.  I went into my Datasource and set Prometheus as 'default', re-uploaded your dashboard and now it was working well.  Perhaps something for you to consider just in case others come across this issue and don't have the Prometheus datasource set as default ?

I have tested the package and dashboard with three separate rhel8 openscap scans.  

One issue I found was that when selecting different Targets, the bar gauge "Failed Rules by Severity" columns would move around to differing positions ?  With my testing, it seemed like 'high' was the only one moving ?  Could you please fix the positions to be high, medium, low, unknown ?

Aside from that, some other cosmetic changes which users could possibly do for themselves, but if you're interested to make them, I would suggest :
- the pie chart; add the value along side percentage in both the 'legend' and on the donut
- the bar gauge; change the 'unknown' colour to 'grey' (previously I did suggest blue, but grey seems more appropriate for unknown :)

As with running the 'scapinoculars' package, for now, I run it as standalone and couldn't see how to set the reports variable ? so it defaults to ./resources/reports.  Could you please advise how to run the package and manually set the reports directory ?

That's all I have for now.

THANKS AGAIN !

**Reply by jritter** [reply]
Hi @levp7 

Thanks for your feedback. I'll have a look at your suggestions regarding the dashboard. I know the issue with the positioning of the Failed rules by severity bars, and so far I haven't found a way how to sort this permanently.

The other things I should be able to address if I find a timeslot to work on it.

Regarding the reports directory.

As per now, you can configure this using an environment variable called REPORT_DIR:

```
export REPORT_DIR=/path/to/your/reports
./scapinoculars
```

Thanks again for your feedback!
